tests/utils/features/query_features.py

```diff
@@ -0,0 +1,450 @@
+        for timestamp in os.listdir(self.dir):
+            timestamp_path = os.path.join(self.dir, timestamp)
+            if not os.path.isdir(timestamp_path):
+                continue
+            for filename in os.listdir(timestamp_path):
+                if filename.endswith('.json'):
+                    system = self.__get_filename_without_last_ext(filename)
+                    dictionary[timestamp].append(system)
+        return [{"timestamp": entry[0], "systems": entry[1]} for entry in sorted(dictionary.items(), reverse=True)]
+
+    def get_systems(self, timestamp: str, systems: list[str] = None) -> Iterable[SystemFeatures]:
+        timestamp_dir = os.path.join(self.dir, timestamp)
+        if not os.path.isdir(timestamp_dir):
+            raise RuntimeError(
+                f'timestamp {timestamp} not present in dir {self.dir}')
+        for filename in os.listdir(timestamp_dir):
+            if filename.endswith('.json') and (not systems or self.__get_filename_without_last_ext(filename) in systems):
+                with open(os.path.join(timestamp_dir, filename), 'r', encoding='utf-8') as f:
+                    yield json.load(f)
+
+    def get_single_json(self, timestamp: str, system: str) -> SystemFeatures:
+        sys_path = os.path.join(self.dir, timestamp, system+'.json')
+        if not os.path.exists(sys_path):
+            raise RuntimeError(f'system file not found {sys_path}')
+        with open(sys_path, 'r', encoding='utf-8') as f:
+            return json.load(f)
+
+
+def consolidate_system_features(system_json: SystemFeatures, include_tasks: Iterable[TaskId] = None, exclude_tasks: Iterable[TaskId] = None) -> dict[str, list[dict[str, Any]]]:
+    '''
+    Given a dictionary of system feature data, consolidates features across
+    all tests into one feature data dictionary.
+
+    :param system_json: a SystemFeatures dictionary
+    :param include_tasks: if not None, will only consolidate tasks present in this list
+    :param exclude_tasks: if not None, will not consolidate tasks present in this list
+    :returns: a dictionary with only feature data
+    '''
+    features = defaultdict(list)
+    for test in system_json['tests']:
+        if include_tasks is not None and TaskIdVariant(test['suite'], test['task_name'], test['variant']) not in include_tasks:
+            continue
+        if exclude_tasks is not None and TaskIdVariant(test['suite'], test['task_name'], test['variant']) in exclude_tasks:
+            continue
+
+        for feature_name in test.keys():
+            if feature_name not in KNOWN_FEATURES:
+                continue
+            for feature in test[feature_name]:
+                if feature not in features[feature_name]:
+                    features[feature_name].append(feature)
+    return features
+
+
+def minus(first: dict[str, list], second: dict[str, list]) -> dict:
+    '''
+    Creates a new dictionary of first - second calculated on values.
+
+    Ex: 
```

Comment:
Can you please make this into an executable documentation (python interactive shell copy-paste). This can be easily tested with built in tools.